src/compiler/utilities.ts

```diff
@@ -1904,12 +1924,32 @@ namespace ts {
 
     /// Given a BinaryExpression, returns SpecialPropertyAssignmentKind for the various kinds of property
     /// assignments we treat as special in the binder
-    export function getAssignmentDeclarationKind(expr: BinaryExpression): AssignmentDeclarationKind {
+    export function getAssignmentDeclarationKind(expr: BinaryExpression | CallExpression): AssignmentDeclarationKind {
         const special = getAssignmentDeclarationKindWorker(expr);
         return special === AssignmentDeclarationKind.Property || isInJSFile(expr) ? special : AssignmentDeclarationKind.None;
     }
 
-    function getAssignmentDeclarationKindWorker(expr: BinaryExpression): AssignmentDeclarationKind {
+    export function isBindableObjectDefinePropertyCall(expr: CallExpression): expr is BindableObjectDefinePropertyCall {
+            return !(length(expr.arguments) !== 3 ||

```

Comment:
`! ... !== .. ||` ? Why not write it as `... === ... &&` ?

Reply:
Also, I thought you said the third one was optional? I guess we won't be able to get the type in that case, but we could still bind the property with type any.

Reply:
It's not optional, it just has a few different shapes it can take.